CodeConverter/CSharp/QueryConverter.cs

```diff
@@ -33,20 +33,25 @@ public QueryConverter(CommonConversions commonConversions, SemanticModel semanti
         public async Task<CSharpSyntaxNode> ConvertClausesAsync(SyntaxList<VBSyntax.QueryClauseSyntax> clauses)
         {
             var vbBodyClauses = new Queue<VBSyntax.QueryClauseSyntax>(clauses);
-            var vbStartClause = vbBodyClauses.Dequeue();
-            var agg = vbStartClause as VBSyntax.AggregateClauseSyntax;
-            if (agg != null) {
+            var vbStartClause = vbBodyClauses.Peek();
+            CSSyntax.FromClauseSyntax fromClauseSyntaxFromAggregate = null;
+            if (vbStartClause is VBSyntax.AggregateClauseSyntax agg) {
+                vbBodyClauses.Dequeue();
```

Comment:
Essentially treats aggregate as a special case, then everything else (including the from clause) is handled in the main loop.